examples/door-lock-app/linux/src/LockEndpoint.cpp

```diff
@@ -0,0 +1,348 @@
+    user.userUniqueId   = userInDb.userUniqueId;
+    user.userType       = userInDb.userType;
+    user.credentialRule = userInDb.credentialRule;
+    user.createdBy      = userInDb.createdBy;
+    user.lastModifiedBy = userInDb.lastModifiedBy;
+
+    ChipLogDetail(
+        Zcl,
+        "Found occupied user "
+        "[endpoint=%d,adjustedIndex=%hu,name=\"%*.s\",credentialsCount=%zu,uniqueId=%x,type=%" PRIu8 ",credentialRule=%" PRIu8 ","
+        "createdBy=%d,lastModifiedBy=%d]",
+        mEndpointId, adjustedUserIndex, static_cast<int>(user.userName.size()), user.userName.data(), user.credentials.size(),
+        user.userUniqueId, to_underlying(user.userType), to_underlying(user.credentialRule), user.createdBy, user.lastModifiedBy);
+
+    return true;
+}
+
+bool LockEndpoint::SetUser(uint16_t userIndex, chip::FabricIndex creator, chip::FabricIndex modifier,
+                           const chip::CharSpan & userName, uint32_t uniqueId, DlUserStatus userStatus, DlUserType usertype,
+                           DlCredentialRule credentialRule, const DlCredential * credentials, size_t totalCredentials)
+{
+    ChipLogProgress(Zcl,
+                    "Door Lock App: LockEndpoint::SetUser "
+                    "[endpoint=%d,userIndex=%" PRIu16 ",creator=%d,modifier=%d,userName=\"%*.s\",uniqueId=%" PRIx32
+                    ",userStatus=%" PRIu8 ",userType=%" PRIu8 ","
+                    "credentialRule=%" PRIu8 ",credentials=%p,totalCredentials=%zu]",
+                    mEndpointId, userIndex, creator, modifier, static_cast<int>(userName.size()), userName.data(), uniqueId,
+                    to_underlying(userStatus), to_underlying(usertype), to_underlying(credentialRule), credentials,
+                    totalCredentials);
+
+    uint16_t adjustedUserIndex = static_cast<uint16_t>(userIndex - 1);
+    if (adjustedUserIndex > mLockUsers.size())
+    {
+        ChipLogError(Zcl, "Cannot set user - index out of range [endpoint=%d,index=%d,adjustedUserIndex=%" PRIu16 "]", mEndpointId,
+                     userIndex, adjustedUserIndex);
+        return false;
+    }
+
+    auto & userInStorage = mLockUsers[adjustedUserIndex];
+
+    if (userName.size() > DOOR_LOCK_MAX_USER_NAME_SIZE)
+    {
+        ChipLogError(Zcl, "Cannot set user - user name is too long [endpoint=%d,index=%d,adjustedUserIndex=%" PRIu16 "]",
+                     mEndpointId, userIndex, adjustedUserIndex);
+        return false;
+    }
+
+    if (totalCredentials > sizeof(DOOR_LOCK_MAX_CREDENTIALS_PER_USER))
+    {
+        ChipLogError(Zcl,
+                     "Cannot set user - total number of credentials is too big [endpoint=%d,index=%d,adjustedUserIndex=%" PRIu16
+                     ",totalCredentials=%zu]",
+                     mEndpointId, userIndex, adjustedUserIndex, totalCredentials);
+        return false;
+    }
+
+    chip::Platform::CopyString(userInStorage.userName, sizeof(userInStorage.userName), userName);
```

Comment:
You can actually do:
```suggestion
    chip::Platform::CopyString(userInStorage.userName, userName);
```
and CopyString will figure out the size of the first arg is its sizeof itself.

Reply:
Fixed